### PR TITLE
add stdio redirection example

### PR DESCRIPTION
**Pull request by sevenautumns**
### 🚀 Features

- *(hypervisor)* Add partition idle logic
- Add stdio redirection example

### 🚜 Refactor

- Remove unused dependencies

### CODE REVIEW

**Comment by wucke13** [comment]
@sevenautumns please rebase

**Review by wucke13 — APPROVED** [review]

File: examples/redirect_stdio/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        let stdin = std::io::stdin();
+        for line in stdin.lock().lines() {
+            println!("{}", line.unwrap())
+        }
+        println!("Finished reading stdin to stdout");
+
+        eprintln!("Error was encountered: None");
+        eprintln!("But it was printed to stderr");
+
+        // TODO wait for https://github.com/DLR-FT/a653rs/issues/22 to be fixed
```
Comment (dadada) [inline examples/redirect_stdio/src/main.rs:69]: Steht als completed drin.

File: flake.nix [hunk 2]
```diff
@@ -71,6 +71,18 @@
             name = "hello_part_no_macros";
             partitions = [ "hello_part_no_macros" ];
           }
+          {
+            name = "redirect_stdio";
+            partitions = [ "redirect_stdio" ];
+            preRun = ''
+              tmpdir=$(mktemp -d -t redirect_stdio.XXXXXXXXXX)
```
Comment (dadada) [inline flake.nix:78]: Should trap to remove the directory automatically on SIGTERM etc.

Comment (dadada) [reply flake.nix:78]: Or even better just put the output somewhere in the current working directory, incase we want to inspect it. No reason to put it in tmpfs?

File: examples/redirect_stdio/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+use anyhow::Result;
+use log::LevelFilter;
+
+fn replace_stdio<T: AsRawFd, U: AsRef<Path>>(stdio: T, new: U, write: bool) -> Result<()> {
+    let new = OpenOptions::new()
+        .write(write)
+        .read(!write)
+        .truncate(write)
+        .open(new)?;
+    nix::unistd::dup2(new.as_raw_fd(), stdio.as_raw_fd())?;
```
Comment (dadada) [inline examples/redirect_stdio/src/main.rs:17]: Can you close stdin afterward?

**Review by dadada — CHANGES_REQUESTED** [review]
I'm against using tmpfs in the example. It is more complicated than needed.

**Review by dadada — APPROVED** [review]

**Pull request merged**